web/controllers/v1/bundles_controller.ex

```diff
@@ -37,4 +39,114 @@ defmodule Cog.V1.BundlesController do
     end
   end
 
+  def create(conn, _) do
+    result = with {:ok, config} <- get_config(conn),
+                  :ok           <- validate_config(config),
+               do: persist(config)
+
+    case result do
+      {:ok, bundle} ->
+        bundle = Repo.preload(bundle, :commands)
+        conn
+        |> put_status(:created)
+        |> put_resp_header("location", bundles_path(conn, :show, bundle))
+        |> render("show.json", %{bundle: bundle})
+      {:error, err} ->
+        send_failure(conn, err)
+    end
+  end
+
+  #### BUNDLE CREATION HELPERS ####
+
+  # First let's grab the config
+  defp get_config(conn) do
+    case conn.params do
```

Comment:
Is `conn.params` different from the `params` (second arg) passed into the controller action? If not it might be nice to do this bit in the action function head and then you only have to send `params` to this function instead of the entire `conn`.